### PR TITLE
Fix `PropertyGraph.renumber_*_by_type` with only default types

### LINKED ISSUE #3058: [FIX]: MGPropertyGraph renumber_vertices_by_type fails when type is unspecified ('')
### Version

23.02

### Which installation method(s) does this occur on?

Docker, Conda, Pip, Source

### Describe the bug.

Called `add_vertex_data` without specifying type.  Called `renumber_vertices_by_type`.  An `IndexError` was thrown.

```
IndexError: string index out of range
```

### Minimum reproducible example

```shell
pG = MGPropertyGraph()
    pG.add_edge_data(
        dask_cudf.from_cudf(
            cudf.DataFrame(
                {
                    "src": cupy.array([0, 0, 1, 2, 2, 3], dtype="int32"),
                    "dst": cupy.array([1, 2, 4, 3, 4, 1], dtype="int32"),
                }
            ),
            npartitions=2,
        ),
        vertex_col_names=["src", "dst"],
    )

    pG.add_vertex_data(
        dask_cudf.from_cudf(
            cudf.DataFrame(
                {
                    "prop1": [100, 200, 300, 400, 500],
                    "prop2": [5, 4, 3, 2, 1],
                    "id": cupy.array([0, 1, 2, 3, 4], dtype="int32"),
                }
            ),
            npartitions=2,
        ),
        vertex_col_name="id"
    )

    pG.renumber_vertices_by_type()
```
```


### Relevant log output

```shell
/opt/conda/envs/rapids/lib/python3.9/site-packages/cugraph/dask/structure/mg_property_graph.py:1313: in renumber_vertices_by_type
    df = df.reset_index().sort_values(by=TCN)
/opt/conda/envs/rapids/lib/python3.9/contextlib.py:79: in inner
    return func(*args, **kwds)
/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_cudf/core.py:225: in sort_values
    df = sorting.sort_values(
/opt/conda/envs/rapids/lib/python3.9/contextlib.py:79: in inner
    return func(*args, **kwds)
/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_cudf/sorting.py:272: in sort_values
    divisions = quantile_divisions(df, by, npartitions)
/opt/conda/envs/rapids/lib/python3.9/contextlib.py:79: in inner
    return func(*args, **kwds)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

df = <dask_cudf.DataFrame | 22 tasks | 2 npartitions>, by = ['_TYPE_'], npartitions = 2

    @_dask_cudf_nvtx_annotate
    def quantile_divisions(df, by, npartitions):
        qn = np.linspace(0.0, 1.0, npartitions + 1).tolist()
        divisions = _approximate_quantile(df[by], qn).compute()
        columns = divisions.columns
    
        # TODO: Make sure divisions are correct for all dtypes..
        if (
            len(columns) == 1
            and df[columns[0]].dtype != "object"
            and not is_categorical_dtype(df[columns[0]].dtype)
        ):
            dtype = df[columns[0]].dtype
            divisions = divisions[columns[0]].astype("int64")
            divisions.iloc[-1] += 1
            divisions = sorted(
                divisions.drop_duplicates().astype(dtype).to_arrow().tolist(),
                key=lambda x: (x is None, x),
            )
        else:
            for col in columns:
                dtype = df[col].dtype
                if dtype != "object":
                    divisions[col] = divisions[col].astype("int64")
                    divisions[col].iloc[-1] += 1
                    divisions[col] = divisions[col].astype(dtype)
                else:
                    divisions[col].iloc[-1] = chr(
>                       ord(divisions[col].iloc[-1][0]) + 1
                    )
E                   IndexError: string index out of range

/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_cudf/sorting.py:222: IndexError
```
```


### Environment details

```shell
Standard environment
```

### Other/Misc.

n/a

### Code of Conduct

- [X] I agree to follow cuGraph's Code of Conduct
- [X] I have searched the [open bugs](https://github.com/rapidsai/cugraph/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by eriknw**
Fixes #3058

### CODE REVIEW

File: python/cugraph/cugraph/dask/structure/mg_property_graph.py [hunk 1]
```diff
@@ -1540,8 +1544,8 @@ def renumber_edges_by_type(self, prev_id_column=None):
 
         # FIXME DASK_CUDF: https://github.com/rapidsai/cudf/issues/11795
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/dask/structure/mg_property_graph.py:1644]: Unfortunate this bug has not been fixed yet

File: python/cugraph/cugraph/structure/property_graph.py [hunk 2]
```diff
@@ -1982,6 +1982,7 @@ def renumber_vertices_by_type(self, prev_id_column=None):
             ].astype(cat_dtype)
 
         index_dtype = self.__vertex_prop_dataframe.index.dtype
+        # Should we avoid `sort_values` if we know there is only one type?
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/structure/property_graph.py:2077]: The answer is yes, we shouldn't call sort_values if there is only one type.  This should match the MG behavior.

**Review by alexbarghi-nv — CHANGES_REQUESTED** [review]
Just 1 minor change request.

File: python/cugraph/cugraph/dask/structure/mg_property_graph.py (hunk 1 above)
Comment (eriknw) [reply python/cugraph/cugraph/dask/structure/mg_property_graph.py:1645]: Indeed. I should have a better way to work around this issue in the next release 🤞

File: python/cugraph/cugraph/structure/property_graph.py (hunk 2 above)
Comment (eriknw) [reply python/cugraph/cugraph/structure/property_graph.py]: Done.

**Review by alexbarghi-nv — APPROVED** [review]
👍

**Pull request merged**